catt/controllers.py

```diff
@@ -61,24 +61,24 @@ def all_info(self):
 ]
 
 
-def get_chromecasts() -> List[pychromecast.Chromecast]:
-    devices = pychromecast.get_chromecasts()
+def get_chromecasts_and_ips(names: Optional[List[str]] = None) -> List[Tuple[pychromecast.Chromecast, str]]:
+    if names:
+        services, browser = pychromecast.discovery.discover_listed_chromecasts(friendly_names=names)
+    else:
+        services, browser = pychromecast.discovery.discover_chromecasts()
+    pychromecast.stop_discovery(browser)
+    return [(pychromecast.get_chromecast_from_service(s, browser.zc), s[4]) for s in services]
+
+
+def get_chromecasts(names: Optional[List[str]] = None) -> List[pychromecast.Chromecast]:
+    devices = [d for d, _ in get_chromecasts_and_ips(names=names)]
```

Comment:
This approach is a bit quirky, but it enables us to keep the `Zeroconf` instance inside of `get_chromecasts_and_ips`. Which also means that we don't need to import `zeroconf` only to keep `mypy` happy.

Reply:
Looks good to me.